app/controllers/queue_items_controller.rb

```diff
@@ -32,4 +43,20 @@ def create_queue_item(video)
   def queue_item_not_exists?(video)
     !current_user.queue_items.find_by video: video
   end
+
+  def update_queue_positions
+    ActiveRecord::Base.transaction do
+      params[:queue_items].each do |queue_item|
+        QueueItem.find(queue_item['id']).update!(position: queue_item['position'])
```

Comment:
If current user sending the update is not the one who owns queue item then this will fire. What if we don't want that to happen.

Preferred style is to store queue item under a variable too, and work with that on next line.